models/DataObject/ClassDefinition/Data.php

```diff
@@ -1199,6 +1199,102 @@ public function getUnique()
         return false;
     }
 
+    /**
+     *  @internal
+     *
+     * @param mixed $data
+     * @param DataObject\Concrete|DataObject\Localizedfield|DataObject\Objectbrick\Data\AbstractData|DataObject\Fieldcollection\Data\AbstractData $container
+     * @param array $params
+     *
+     * @throws \Exception
+     */
+    protected function setDataToObject($data, $container, $params = [])
```

Comment:
can you explain me why this complicated function has to be in the "sort-of" abstract parent class for all object data-types? Can't it be extracted to corresponding types?

Also, I don't quite understand where this comes from, shouldn't this only change the behaviour of validation and nothing else?

Reply:
A scenario for introducing this method is that you set duplicate relations to object and then save it, which then filters the duplicate data during save process so then filtered relations data should be set back to object to avoid having different information between object and database. 

The reason it is in Data class as data could be anything not just relation types and also it is the set counterpart of function `getDataFromObjectParam` which is called before applying filters.

Reply:
so basically it is responsible for setting the filtered items back to the object after/on save? (brick, localized-fields,...)

If that is the case, I think you should completely re-think this solution, it is way to un-generic and should be done within the Object Data Tag itself. And it doesn't belong where it currently sits.

It is only used within the `filterMultipleAssignments` currently, where it is the wrong place as well. Every Object Data Tag **should** filter/handle that themself. The `filter` function already is called within the `AdvancedRelation` and obviously `AbstractRelation`, why not just call a setter with the filtered data there?! Then you would get completely rid of that function. So, the `filter` function returns only the allowed data-set, the Object Data Tag then persists said data and set's it back to the object.

Reply:
The thing is that the data-type can be used in many different contexts/containers, you'd have to duplicate this code again and again, that's why we thought it makes sense to make this generic. It's marked internally anyway, if we don't see any further use, we're going to kick it with v7 together with this filtering stuff.

Reply:
Alright then. Keeping it as a temp solution until Pimcore 7 is fine with me.